src/engine/engine.cc

```diff
@@ -23,10 +23,30 @@
  * \brief Implementation of engine.
  */
 #include <mxnet/engine.h>
+#include <fenv.h>
```

Comment:
QQ: do we follow any rules (like alphabetic order of library name) to include libraries?